src/Detector/Hugo/HugoDetector.cs

```diff
@@ -156,17 +156,7 @@ private bool IsHugoTomlFile(ISourceRepo sourceRepo, params string[] subPaths)
             TomlTable tomlTable = null;
 
             var relativeFilePath = Path.Combine(subPaths);
-
-            try
-            {
-                tomlTable = ParserHelper.ParseTomlFile(sourceRepo, relativeFilePath);
-            }
-            catch (FailedToParseFileException ex)
-            {
-                _logger.LogError(ex, $"An error occurred when trying to parse file '{relativeFilePath}'.");
-                return false;
-            }
-
+            tomlTable = ParserHelper.ParseTomlFile(sourceRepo, relativeFilePath);
```

Comment:
In CLI `CommandBase`, we only output specific messages from `InvalidUsageException` so we don't want to early catch `FailedToParseFileException ` in detector here.

Reply:
so where are we placing the generalexception in terms of order? can we add few tests where we can validate this ... like exception1 gets caught before exception2 and exception2 gets caught before exception3 and so on

Reply:
Now we're having Detector level exceptions and Platform level exceptions, we can validate each of them in DetectorTests and PlatformTests. The order will be like, a detector level exception throws and we catch it in a platform and throws a platform level exception there.